core/src/main/java/io/grpc/internal/SpiffeUtil.java

```diff
@@ -119,4 +286,34 @@ public String getPath() {
     }
   }
 
+  /**
+   * Represents a Trust Bundle inspired by SPIFFE Bundle Format standard. Only trust domain's
+   * sequence numbers and x509 certificates are supported.
+   * @see <a href="https://github.com/spiffe/spiffe/blob/main/standards/SPIFFE_Trust_Domain_and_Bundle.md#4-spiffe-bundle-format">Standard</a>
+   */
+  public static final class SpiffeBundle {
+
+    private final ImmutableMap<String, Long> sequenceNumbers;
+
+    private final ImmutableMap<String, ImmutableList<X509Certificate>> bundleMap;
+
+    private SpiffeBundle(Map<String, Long> sequenceNumbers,
+        Map<String, List<X509Certificate>> trustDomainMap) {
+      this.sequenceNumbers = ImmutableMap.copyOf(sequenceNumbers);
```

Comment:
micro-nit: "this." is redundant here and can be removed.

Reply:
In this particular case it's mandatory - the arg name equals to class field name.

Reply:
Facepalm, sorry you're totally correct. :)